src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +358,507 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
+                                     vm={vm} />}
+                </Form>
+                <AdditionalOptions cacheMode={diskParams.cacheMode}
+                                   device={diskParams.device}
+                                   idPrefix={idPrefix}
+                                   onValueChanged={onValueChanged}
+                                   busType={diskParams.busType}
+                                   serial={diskParams.serial}
+                                   validationFailed={validationFailed}
+                                   supportedDiskBusTypes={supportedDiskBusTypes} />
+            </>
+        );
     }
 
-    dialogErrorSet(text, detail) {
-        this.setState({ dialogError: text, dialogErrorDetail: detail });
-    }
+    return (
+        <Modal position="top" variant="medium" id={`${idPrefix}-dialog-modal-window`} isOpen onClose={Dialogs.close}
+               title={isMediaInsertion ? _("Insert disc media") : _("Add disk")}
+               footer={
+                   <AddDiskModalFooter
+                     customDiskVerificationFailed={customDiskVerificationFailed}
+                     customDiskVerificationMessage={customDiskVerificationMessage}
+                     dialogLoading={dialogLoading}
+                     dialogErrorSet={dialogErrorSet}
+                     disk={disk}
+                     diskParams={diskParams}
+                     idPrefix={idPrefix}
+                     isMediaInsertion={isMediaInsertion}
+                     mode={mode}
+                     setValidationFailed={setValidationFailed}
+                     storagePoolName={storagePoolName}
+                     storagePools={storagePools}
+                     verificationInProgress={verificationInProgress}
+                     vm={vm}
+                     vms={vms}
+                   />
+               }
+        >
+            {dialogError && <ModalError dialogError={dialogError} dialogErrorDetail={dialogErrorDetail} />}
+            {defaultBody}
+        </Modal>
+    );
+};
+
+const AddDiskModalFooter = ({
+    customDiskVerificationFailed,
+    customDiskVerificationMessage,
+    dialogErrorSet,
+    dialogLoading,
+    diskParams,
+    idPrefix,
+    isMediaInsertion,
+    mode,
+    setValidationFailed,
+    storagePoolName,
+    storagePools,
+    verificationInProgress,
+    vm,
+    vms,
+}) => {
+    const [addDiskInProgress, setAddDiskInProgress] = useState(false);
+    const Dialogs = useDialogs();
+
+    const validateParams = () => {
+        const validationFailed = {};
+        const storagePoolType = diskParams.storagePool?.type;
 
-    onInsertClicked() {
-        const Dialogs = this.context;
-        const close = Dialogs.close;
-        const { vm } = this.state;
-        const { disk } = this.props;
+        if (mode !== CUSTOM_PATH && !storagePoolName)
+            validationFailed.storagePool = _("Please choose a storage pool");
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231113-155652-97d90b8c-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)